rules/preprocessing_methyldackel.py

```diff
@@ -0,0 +1,199 @@
+        minqual = int(config['general']
+                      ['methylation-calling']['minimum-quality'])
+    log:
+        DIR_methcall + "{sample}.deduped.methyldackel_calls.log"
+    message: fmt("Extract methylation calls from bam file using MethylDackel for sample {{sample}} and protocol {params.protocol}")
+    shell:
+        nice("methyldackel",
+             ["extract", "{input.genome}", "{input.bamfile}",
+              "-o {params.prefix}", "-@ {params.threads}", "{params.keepDups}",
+              "--methylKit", "--CHH", "--CHG", "-q {params.minqual}"],
+             ("{log}"))
+
+
+# ==========================================================================================
+# Extract methylation bias with methylDackel
+#
+
+rule methyldackel_mbias:
+    input:
+        bamfile = DIR_sorted + "{sample}.bwameth.sorted.markdup.bam",
+        genome = GENOMEFILE
+    output:
+        txt = DIR_methcall + "methylDackel/" + "{sample}_mbias_methyldackel.txt",
+        p1 = DIR_methcall + "methylDackel/"+ "{sample}_mbias_OB.svg",
+        p2 = DIR_methcall + "methylDackel/" + "{sample}_mbias_OT.svg"
+    params:
+        threads = config['execution']['rules']['methyldackel_extract']['threads'],
+        prefix = DIR_methcall + "methylDackel/" + "{sample}_mbias",
+        protocol = lambda wc: protocol(wc),
+        keepDups = keepDups("{params.protocol}"),
+        minqual = int(config['general']
+                      ['methylation-calling']['minimum-quality'])
+    log:
+        DIR_methcall + "{sample}.methyldackel_mbias.log"
+    message: fmt("Calculate methylation bias using MethylDackel for sample {{sample}}.")
+    shell:
+        nice("methyldackel",
+             ["mbias", "{input.genome}", "{input.bamfile}",
+              "{params.prefix}", "{params.keepDups}",
+              "-@ {params.threads}", "--txt",
+              "-q {params.minqual}", "> {output.txt}",
+              "2> {log}"])
+
+
+# ==========================================================================================
+# Extract methylation bias with methylDackel
+#
+
+rule methyldackel_cytosine_report:
+    input:
+        bamfile = DIR_sorted + "{sample}.bwameth.sorted.markdup.bam",
+        genome = GENOMEFILE
+    output:
+        DIR_methcall + "methylDackel/" + "{sample}_methyldackel.cytosine_report.txt"
+    params:
+        threads = config['execution']['rules']['methyldackel_extract']['threads'],
+        prefix = DIR_methcall + "methylDackel/" + "{sample}_methyldackel",
+        protocol = lambda wc: protocol(wc),
+        keepDups = keepDups("{params.protocol}"),
+        minqual = int(config['general']
+                      ['methylation-calling']['minimum-quality'])
+    log:
+        DIR_methcall + "{sample}.methyldackel_cytosine_report.log"
+    message: fmt("Extract cytosine report from bam file using MethylDackel for sample {{sample}} and context {params.context}")
+    shell:
+        nice("methyldackel",
```

Comment:
In the configure file you search for "MethylDackel"; what's the actual command name?